src/allotropy/parsers/chemometec_nucleoview/nucleoview_structure.py

```diff
@@ -1,66 +1,77 @@
+    Data,
+    Measurement,
+    MeasurementGroup,
+    Metadata,
+)
+from allotropy.named_file_contents import NamedFileContents
 from allotropy.parsers.chemometec_nucleoview.constants import (
     DEFAULT_ANALYST,
     DEFAULT_EPOCH_TIMESTAMP,
     DEFAULT_MODEL_NUMBER,
+    NUCLEOCOUNTER_DETECTION_TYPE,
+    NUCLEOCOUNTER_DEVICE_TYPE,
+    NUCLEOCOUNTER_SOFTWARE_NAME,
 )
-from allotropy.parsers.utils.pandas import map_rows, SeriesData
+from allotropy.parsers.chemometec_nucleoview.nucleoview_reader import NucleoviewReader
+from allotropy.parsers.utils.pandas import df_to_series_data, map_rows, SeriesData
+from allotropy.parsers.utils.uuids import random_uuid_str
+
 
+def _create_metadata(data: SeriesData, file_name: str) -> Metadata:
+    return Metadata(
+        file_name=file_name,
+        model_number=data.get(str, "Instrument type", DEFAULT_MODEL_NUMBER),
+        equipment_serial_number=data.get(str, "Instrument s/n"),
+        software_name=NUCLEOCOUNTER_SOFTWARE_NAME,
+        software_version=data.get(str, "Application SW version"),
+        device_type=NUCLEOCOUNTER_DEVICE_TYPE,
+        detection_type=NUCLEOCOUNTER_DETECTION_TYPE,
+    )
 
-@dataclass
-class Row:
-    model_number: str
-    equipment_serial_number: str | None
-    software_version: str | None
-    analyst: str
-    timestamp: str
-    sample_identifier: str
-    multiplication_factor: float | None
-    viability_percent: float
-    live_cell_count: JsonFloat
-    dead_cell_count: JsonFloat
-    total_cell_count: JsonFloat
-    estimated_cell_diameter: JsonFloat
 
-    @staticmethod
-    def create(data: SeriesData) -> Row | None:
-        if data.get(float, "Viability (%)", validate=SeriesData.NOT_NAN) is None:
-            return None
+def _create_measurement_groups(data: SeriesData) -> MeasurementGroup:
+    timestamp = data.get(str, "Date time")
+    if timestamp:
+        offset = data.get(str, "Time zone offset", "0")
+        timestamp = pd.to_datetime(
+            f"{timestamp}{'+' if offset[0] not in {'+', '-'} else ''}{offset}"
+        ).isoformat()
 
-        timestamp = data.get(str, "Date time")
-        if timestamp:
-            offset = data.get(str, "Time zone offset", "0")
-            timestamp = pd.to_datetime(
-                f"{timestamp}{'+' if offset[0] not in {'+', '-'} else ''}{offset}"
-            ).isoformat()
+    return MeasurementGroup(
+        analyst=data.get(str, "Operator", DEFAULT_ANALYST),
+        measurements=[
+            Measurement(
+                measurement_identifier=random_uuid_str(),
+                timestamp=timestamp or DEFAULT_EPOCH_TIMESTAMP,
+                sample_identifier=data[str, "Image"].split("-")[3],
+                cell_density_dilution_factor=data.get(float, "Multiplication factor"),
+                viability=data[float, "Viability (%)"],
+                # Cell counts are measured in cells/mL, but reported in millions of cells/mL
```

Comment:
I see this is happening a lot ... is it worth putting a helper that automatically divides? up to you